src/Plug.vala

```diff
@@ -1,191 +1,143 @@
+        supported_settings.set ("security/screensaver", LOCKING);
+    }
 
+    public override Gtk.Widget get_widget () {
+        if (paned == null) {
             try {
-                var permission = new Polkit.Permission.sync (
+                permission = new Polkit.Permission.sync (
                     "io.elementary.switchboard.security-privacy",
                     new Polkit.UnixProcess (Posix.getpid ())
                 );
-
-                lock_button = new Gtk.LockButton (permission);
-
-                infobar.revealed = false;
-                infobar.get_action_area ().add (lock_button);
-
-                stack.notify["visible-child-name"].connect (() => {
-                    if (permission.allowed == false && stack.visible_child_name == "firewall") {
-                        infobar.revealed = true;
-                    } else {
-                        infobar.revealed = false;
-                    }
-                });
-
-                permission.notify["allowed"].connect (() => {
-                    if (permission.allowed == false && stack.visible_child_name == "firewall") {
-                        infobar.revealed = true;
-                    } else {
-                        infobar.revealed = false;
-                    }
-                });
             } catch (Error e) {
                 critical (e.message);
             }
 
-            tracking = new TrackPanel ();
+            var label = new Gtk.Label (_("Some settings require administrator rights to be changed"));
+
+            var lock_button = new Gtk.LockButton (permission);
+
+            var infobar = new Gtk.InfoBar () {
+                message_type = INFO,
+                revealed = false
+            };
+            infobar.get_content_area ().add (label);
+            infobar.get_action_area ().add (lock_button);
+
+            var tracking = new TrackPanel ();
             var locking = new LockPanel ();
-            firewall = new FirewallPanel ();
-            housekeeping = new HouseKeepingPanel ();
-            location = new LocationPanel ();
+            var firewall = new FirewallPanel (permission);
+            var housekeeping = new HouseKeepingPanel ();
+            var location = new LocationPanel ();
 
+            stack = new Gtk.Stack ();
             stack.add_titled (tracking, HISTORY, _("Privacy"));
             stack.add_titled (locking, LOCKING, _("Locking"));
             stack.add_titled (firewall, FIREWALL, _("Firewall"));
             stack.add_titled (housekeeping, HOUSEKEEPING, _("Housekeeping"));
             stack.add_titled (location, LOCATION, _("Location Services"));
 
+            var box = new Gtk.Box (VERTICAL, 0);
+            box.add (infobar);
+            box.add (stack);
+
             var settings_sidebar = new Granite.SettingsSidebar (stack);
 
-            var paned = new Gtk.Paned (Gtk.Orientation.HORIZONTAL);
-            paned.position = 200;
```

Comment:
was the removal of position here intentional?

Reply:
Yeah, settingssidebar already seems to reserve enough space